### PR TITLE
Jetpack: Add Jetpack Share Buttons block.

### LINKED ISSUE #34050: Create Sharing Beta (dummy) block
As a first step for 'Share this' feature we are creating a Sharing Beta block. At this state it will work within editor, but links in this Post view won't work properly yet.

### PR DESCRIPTION
**Pull request by heydemoura**
Fixes: https://github.com/Automattic/jetpack/issues/34050

#### Changes proposed in this Pull Request:

This Pull Request originated from this Hack Day project (p1HpG7-iVl-p2), which our team couldn't finish at that day specifically, but we combined a lot of knowledge into building a prototype, which could then be picked up in the future to build the actual solution. 

On March 2023 Hack Week, I picked up on this project to rise attention to it, since I wish for this feature myself. This is not yet finished, but it is on a much better stance and on a simple implementation path. It still relies on sharedaddy on the backend.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p1HpG7-iVl-p2
pebfNN-6v-p2

#### Testing instructions:

* Run the [extensions build ](https://github.com/Automattic/jetpack/tree/trunk/projects/plugins/jetpack/extensions)
* Edit any post (new or existing)
* Add a block, should be called **Sharing Buttons**
* Clicking the buttons should select services, save the post after selecting a few
* Preview/Visit the post to see the sharing buttons rendered where you added the block.

image

image

### CODE REVIEW

**Comment by oskosk** [comment]
Pushed a rebase (done with `git rebase trunk` so I had to force-push).

**Comment by oskosk** [comment]
Pushed another rebase (with force-push again)

**Comment by andrii-lysenko** [comment]
After discussing this with @jeherve we decided to keep this PR simpler and would push this as a Beta Block. It will work in editor, but links in the view are not completely working, as we would need to wire them up with Sharedaddy, as well as with WPCOM endpoints.

Some updates on what's been done recently:
- Refactored the code according to ESLint configs and fixed the namings
- Updated block according to new Jetpack Block Development guidelines (p1HpG7-pF0-p2)
- Added 'discoverability' logic aligned with other blocks that rely on modules
- Cleaned up unused code, we could add this back in future if needed.
- Tested on self-hosted and simple/atomic (there's issue with icons on simple/atomic, but we would iterate on this in the future)

So far I think it is good for review and can be deployed as a first step, not to over-complicate things.

**Comment by kraftbj** [comment]
I'm going to approve this as an iterative step forward. There are a couple of quick things that I noticed from functional testing that I think we should address:

* when the block is in focus, the UI makes sense. I would expect, though, when the block is out of focus, the button look like they would on the front-end. e.g. this doesn't give me a real representation of what it will look like:

2023-11-09 at 15 58

on a brand new site, I added the block. It, nicely, told me to activate the Sharing Buttons feature
* Do we actually need this step? Can we leverage the Sharedaddy code without the module activation? In other words, separate the current `the_content`-based method with the actual functionality of the sharing button?
* We shouldn't display both or at least maybe align them more?
 
2023-11-09 at 16 00

This may be a bad example since the post is literally just the share block, but it's notable that we're inserting the sharing buttons on `the_content` AND in the post itself. Maybe that's okay, but would like us to process it.

* More critically, we can't continue to `the_content` the buttons if they are using the site editor to add them. Here's three (different) sets of sharing buttons—the block in the post, the `the_content` hook that is edited via settings, then a block inserted in to a template for a single post.
2023-11-09 at 16 03

Apologies if this is all already documented, but wanted to note it before approving. :)

**Review by kraftbj — APPROVED** [review]

**Comment by andrii-lysenko** [comment]
@kraftbj Thanks for your review. It's definitely just a step, to start moving forward and these are definitely the problems we would need to solve. 
We would need to find a way to use this alongside sharedaddy. As I understood, original `Share this:` will be removed, and we would automatically put Block instead (we would need to make sure it won't break view of the present sites.)

**Review by jeherve — APPROVED** [review]
Thanks!

Let's get this merged and we can iterate in follow-up PRs! 🚢

File: projects/plugins/jetpack/extensions/blocks/sharing-buttons/block.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"category": "embed",
+	"icon": "<svg viewBox='0 0 24 24' xmlns='http://www.w3.org/2000/svg'><path d='M9 11.8l6.1-4.5c.1.4.4.7.9.7h2c.6 0 1-.4 1-1V5c0-.6-.4-1-1-1h-2c-.6 0-1 .4-1 1v.4l-6.4 4.8c-.2-.1-.4-.2-.6-.2H6c-.6 0-1 .4-1 1v2c0 .6.4 1 1 1h2c.2 0 .4-.1.6-.2l6.4 4.8v.4c0 .6.4 1 1 1h2c.6 0 1-.4 1-1v-2c0-.6-.4-1-1-1h-2c-.5 0-.8.3-.9.7L9 12.2v-.4z' /></svg>",
+	"supports": {
+		"align": false,
+		"alignWide": true,
+		"anchor": false,
+		"customClassName": true,
+		"className": true,
+		"html": false,
+		"multiple": false,
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/sharing-buttons/block.json:19]: Any tehnical blockers letting the block be in the post or template multiple times?

Seems handy if they want to have the sharing e.g. on top of the post, and in the footer.

Even in the middle section for some reason, why not! ;-)

File: projects/plugins/jetpack/extensions/blocks/sharing-buttons/available-services.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	email: { label: 'Email', icon: 'mail' },
+	facebook: { label: 'Facebook', icon: 'facebook' },
+	linkedin: { label: 'LinkedIn', icon: 'linkedin' },
+	reddit: { label: 'Reddit', icon: 'reddit' },
+	x: { label: 'X', icon: 'x' },
+	tumblr: { label: 'Tumblr', icon: 'tumblr' },
+	pinterest: { label: 'Pinterest', icon: 'pinterest' },
+	pocket: { label: 'Pocket', icon: 'pocket' },
+	telegram: { label: 'Telegram', icon: 'telegram' },
+	'jetpack-whatsapp': { label: 'WhatsApp', icon: 'whatsapp' },
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/sharing-buttons/available-services.js:12]: Where's "Press this"?

image

File: projects/plugins/jetpack/extensions/blocks/sharing-buttons/block.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 1,
+	"name": "jetpack/sharing-buttons",
+	"title": "Sharing Buttons",
+	"description": "Sharing Buttons.",
+	"keywords": [ "sharing", "social" ],
+	"version": "12.5.0",
+	"textdomain": "jetpack",
+	"category": "embed",
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/sharing-buttons/block.json:10]: I think you should rethink the category. 

"Related posts" is also in embeds, and "Social icons" in widgets.

Neither category feels like right fit.

**Pull request merged**